src/java.base/unix/native/libjava/io_util_md.c

```diff
@@ -32,6 +32,9 @@
 
 #if defined(__linux__) || defined(_ALLBSD_SOURCE) || defined(_AIX)
 #include <sys/ioctl.h>
+#endif
+
+#if defined(__linux__)
 #include <linux/fs.h>
 #include <sys/stat.h>
```

Comment:
Does Mac OS need `sys/stat.h`, though?

Reply:
No, change is specific to Linux. Please see(https://github.com/openjdk/jdk/commit/69b96f9a1b4a959c6b86f41c2259d9e4d47c8ede).

Reply:
OK, so what you are saying is that the path that references `S_ISBLK` is protected by `BLKGETSIZE64` that is guaranteed to be undefined on OS X? If so, this is (awkwardly) fine.

Reply:
I think Vymon took the code from FileDispatcherImpl_size0, which has been compiling on macOS without issues. I don't mind if we fix the issue with this PR or just back it out and fix it with another PR.

Reply:
I am not the Mac expert, but i checked on my MacOS Laptop and i did not found "BLKGETSIZE64" defined in sys/stat.h.
As Alan already told that i  took the code from FileDispatcherImpl.size0 and followed the same pattern.

Reply:
I will fix in this PR only. i will do the changes for only "io_util_md.c".

Reply:
updated the PR as suggested.